common/src/types/v0/message_bus/nexus.rs

```diff
@@ -179,6 +179,89 @@ pub struct CreateNexus {
     pub managed: bool,
     /// Volume which owns this nexus, if any
     pub owner: Option<VolumeId>,
+    /// Nexus Nvmf Configuration
+    pub config: Option<NexusNvmfConfig>,
+}
+
+/// Nvmf Controller Id Range
+#[derive(Serialize, Deserialize, Debug, Clone, Eq, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct NvmfControllerIdRange(std::ops::Range<u16>);
+impl NvmfControllerIdRange {
+    /// create `Self` with a random minimum controller id
+    pub fn random_min() -> Self {
+        let min = Self::valid_controller_range().start;
+        let max = Self::valid_controller_range().end;
+        let rand_min = u16::min(rand::random::<u16>() + min, max);
+        Self(rand_min .. max)
+    }
+    /// minimum controller id
+    pub fn min(&self) -> u16 {
+        self.0.start
+    }
+    /// maximum controller id
+    pub fn max(&self) -> u16 {
+        self.0.end
+    }
+    fn valid_controller_range() -> std::ops::Range<u16> {
```

Comment:
Nit: This could just be "controller_id_range"

Reply:
Done